### DISCUSSION TITLE
Not able to play large files

### DISCUSSION THREAD

**Discussion by baseliners** [muxinc/media-chrome#979] [Q&A]
Hi - I've been using media-chrome in a Svelte app and am trying to play a few local mp4 videos (H264 codec). It's been great so far but I just noticed that large files (e.g. a 3hr long movie) do not play - i.e. I see a black box and clicking on play just shows a black screen. 

I then used ffmpeg to create clips from this video of varying lengths and found that if I take the first 60 or even 70mins clip, it plays fine. If I take 90mins, I get a black screen but can hear the audio fine. If I take the entire clip, it seems to play (I can see the timecode changing) but no audio or video. 

Based on some research, it looks like html5 video may not work well with very large files and may require some sort of chunking solution. Curious if there's any way to get large files to work properly in media-chrome (with or without chunking). Thanks!

**Comment by luwes** [comment]
I recommend having a read on https://howvideo.works as well.

Your users will have a better experience with a chunked delivery like HLS or DASH.
Media Chrome supports both of them via custom media elements (https://github.com/muxinc/media-elements)
and in addition supports many video platforms like Vimeo, Wistia, YouTube, Cloudflare etc.

That being said chunked is not absolutely necessary. Pseudo streaming with a long MP4 should be possible if the format / compression is correct (use ffmpeg or handbrake), make sure the MOOV atom is at the start of the file (-movflags faststart) and if your server supports 206 partial content.

**Reply by baseliners** [reply]
Thanks for the reply. Few clarifications: 
- my scenario is that the video is local on the hard disk (it's a Svelte app within the Tauri framework for building Mac apps; so I'm trying to play a local video in a Mac app built via Tauri+Svelte). 
- the video format is not something I can control so HLS and DASH are not real options for me
- I did try faststart but it makes no difference for the larger files, I can see the timecode advancing but no video or audio. 

I don't think it's a format/compression issue since as I mentioned, if I create subclips from the main video using ffmpeg, they do work until I get to more than 70mins. 

Pseudo streaming - any pointers or examples for how to do this with media-chrome? Tried a few things but not quite able to get it to work. Thanks!
